server/transaction/manager.go

```diff
@@ -130,10 +145,45 @@ func newTxSession(kv kv.KeyValueStore) (*TxSession, error) {
 	}, nil
 }
 
+func newTxSessionWithMetrics(kv kv.KeyValueStore) (*TxSessionWithMetrics, error) {
+	if kv == nil {
+		return nil, api.Errorf(api.Code_INTERNAL, "session needs non-nil kv object")
+	}
+	return &TxSessionWithMetrics{&TxSession{
+		context: &SessionCtx{},
+		kvStore: kv,
+		state:   sessionCreated,
+		txCtx:   generateTransactionCtx(),
+	}}, nil
+}
+
+func (m *TxSessionWithMetrics) measure(ctx context.Context, name string, f func(ctx context.Context) error) {
```

Comment:
this will be overkill here, once you move it to sessions.go then let's remove it from here.

Reply:
Will do.